### PR TITLE
Terms of service check box on sign up

### PR DESCRIPTION
**Pull request by SanJSp**
A new requirement is to include an explicit terms of service (AGB) checkbox in the sign-up process.

This is the beginning of a solution but I'm stuck. Maybe somebody can help me out here.

The plan is to use the form fields provided by [simple_form](https://github.com/heartcombo/simple_form) to simply add a checkbox for the ToS and verify it in the student model. In the end we also want to enforce ToS agreement for employers aswell, hence the validation will take place in the user model, but lets start small first. The current implementation is oriented at this [post](https://stackoverflow.com/questions/25183094/terms-and-agreement-checkbox-in-simple-form-for/25357938#25357938). Many similar posts( [1](https://stackoverflow.com/questions/42541944/rails-form-for-checkbox-validation-without-record-in-database)) exist on SO. But I don't get it to work correctly.

Currently, when a student checks the ToS checkbox when signing up via "Als Student registrieren", an error raises up. This error states, that the checkbox should be checked, although it was at time of submission.

Here is a screencap of the current frontend, including the error message
image

### CODE REVIEW

File: app/views/students/new.html.erb [hunk 1]
```diff
@@ -6,6 +6,7 @@
     <%= render 'shared/object_errors', object: @student %>
     <%= render 'registration_fields', student: @student, form: f %>
 
+
```
Comment (Dome-GER) [inline app/views/students/new.html.erb:9]: Unrelated change

**Review by Dome-GER — CHANGES_REQUESTED** [review]
Regarding the problem your are facing:

The `terms_of_service` param needs to be permitted in the [StudentsController#student_params](https://github.com/openHPI/hpi-connect-portal/blob/master/app/controllers/students_controller.rb#L144). Otherwise, it will be `nil` for the validation in the model, which results in the error you are seeing.

**Comment by SanJSp** [comment]
Thank you for your feedback @Dome-GER 

I was told that the TOS are only relevant for the employers (not the students - don't ask me why) hence I had to move this functionality to the employer signup flow. Your recommendation worked for the students, but does no longer work for employers. I tried to permit the `:terms_of_service` in the staff members, but that did not work. The error is still the same, when checking the box, I still face the following error message:

image

Maybe you can also give advice on how to make this a nicer error message, everything in one language and without "staff members" would be great!

Thanks again for your time!

File: app/views/staff/_form_fields.html.erb [hunk 2]
```diff
@@ -65,6 +65,21 @@
           </div>
         </div>
       </li>
+      <li class="list-group-item">
+        <div class="row">
+          <div class="col-sm-3">
+            <%= label_field form, :terms_of_service, t("activerecord.attributes.staff.terms_of_service") %>
```
Comment (Dome-GER) [inline app/views/staff/_form_fields.html.erb:71]: At the top of the file, you can see that it is added to the user form:
```
<%= form.simple_fields_for :user, user do |user_form| %>
```

So I assume you have to permit the `:terms_of_service` for the `user_attributes` in `staff_params`, similar to the other user attributes such as `firstname` and `lastname`.

**Review by Dome-GER — CHANGES_REQUESTED** [review]

File: app/views/staff/_form_fields.html.erb (hunk 2 above)
Comment (SanJSp) [reply app/views/staff/_form_fields.html.erb:71]: Did you mean like this in [staff_controller](https://github.com/openHPI/hpi-connect-portal/blob/f83b6882b711c51dbadeca363f2028fd7a51ef9b/app/controllers/staff_controller.rb)?

```ruby
def staff_params
      params.require(:staff).permit(user_attributes: [:firstname, :lastname, :email, :password, :password_confirmation, :terms_of_service])
end
```

unfortunately this did not work. Checking the checkbox still results in the same error as above

Comment (Dome-GER) [reply app/views/staff/_form_fields.html.erb:71]: Oh, now I see why it is still `form`. You want to validate it for `staff` and not the generic `user`.
The validation for `validates_acceptance_of` is still on the `staff` so it will also fail following my comment above.

Comment (Dome-GER) [reply app/views/staff/_form_fields.html.erb:71]: Please ignore my previous comments.

Are you sure this is still an issue? It worked for me locally.

Comment (SanJSp) [reply app/views/staff/_form_fields.html.erb:71]: Unfortunately, it does not on my machine. Still, when I check the checkbox and submit I receive the error “Staff members terms of service muss akzeptiert werden"

**Comment by SanJSp** [comment]
Hey @Dome-GER  I've deployed the changes here to [staging](http://172.16.65.86) (from HPI Network) but it still does not work. When trying to sign up a employer and then checking the ToS checkbox will not successfully sign one up. Does it still work on your machine? Because then it may be because of our browsers?

File: app/views/staff/_form_fields.html.erb (hunk 2 above)
Comment (chrisma) [reply app/views/staff/_form_fields.html.erb:71]: As a first possible step to debug, is the `:terms_of_service` attribute sent as part of the POST request when submitting the form? 
https://stackoverflow.com/questions/15603561/how-can-i-debug-a-http-post-in-chrome

**Comment by Dome-GER** [comment]
Timeout.
